cime/src/drivers/mct/main/prep_glc_mod.F90

```diff
@@ -607,6 +777,156 @@ end subroutine prep_glc_map_one_state_field_lnd2glc
+
+    ! outputs to melt flux calculation
+    index_g2x_Sg_blis = mct_avect_indexra(g2x_gx,'Sg_blis',perrwith='quiet')
+    index_g2x_Sg_blit = mct_avect_indexra(g2x_gx,'Sg_blit',perrwith='quiet')
+    index_g2x_Fogx_qiceho = mct_avect_indexra(g2x_gx,'Fogx_qiceho',perrwith='quiet')
+    index_g2x_Fogx_qicelo = mct_avect_indexra(g2x_gx,'Fogx_qicelo',perrwith='quiet')
+    index_x2g_Fogx_qiceli = mct_avect_indexra(x2g_gx,'Fogx_qiceli',perrwith='quiet')
+    index_x2g_Fogx_qicehi = mct_avect_indexra(x2g_gx,'Fogx_qicehi',perrwith='quiet')
+
+    gsize = mct_aVect_lsize(g2x_gx)
+    write(logunit,*) 'gsize=',gsize
+
+    do n=1,gsize
+      !Extract glc and ocn-sourced coupler fields used as input to compute_melt_fluxes to local arrays...
+
+      ! Fields from the ocean, now on the GLC grid
+      oceanTemperature(n) =               x2g_gx%rAttr(index_x2g_So_blt,n)
+      oceanSalinity(n) =                  x2g_gx%rAttr(index_x2g_So_bls,n)
+      oceanHeatTransferVelocity(n) =      x2g_gx%rAttr(index_x2g_So_htv,n)
+      oceanSaltTransferVelocity(n) =      x2g_gx%rAttr(index_x2g_So_stv,n)
+
+      ! Fields from the ice sheet model (still on the GLC grid)
+      iceTemperature(n) =                 g2x_gx%rAttr(index_g2x_Sg_tbot,n)
+      iceTemperatureDistance(n) =         g2x_gx%rAttr(index_g2x_Sg_dztbot,n)
+      interfacePressure(n) =              g2x_gx%rAttr(index_g2x_Sg_lithop,n)
+      iceFloatingMask(n) =                g2x_gx%rAttr(index_g2x_Sg_icemask_floating,n)
+
+      !... initialize local compute_melt_fluxes output arrays...
+      outInterfaceSalinity(n)     =       0.0_r8
+      outInterfaceTemperature(n)  =       0.0_r8
+      outFreshwaterFlux(n)  =             0.0_r8
+      outOceanHeatFlux(n) =               0.0_r8
+      outIceHeatFlux(n) =                 0.0_r8
+    end do
+
+    !Fyke: hardwire in some *reasonable* values here to test things out.
+      !     do n=1,gsize
+      !       oceanTemperature(n)=0._r8
+      !       oceanSalinity(n)=34.5_r8
+      !       oceanHeatTransferVelocity(n)=1.e-4_r8
+      !       oceanSaltTransferVelocity(n)=3.e-6_r8
+      !       interfacePressure(n)=1000._r8*9.81_r8*918._r8 !lithostatic pressure of 1km of ice
+      !     end do
+
+    !...calculate fluxes...
+    call compute_melt_fluxes(oceanTemperature,&
+                             oceanSalinity,&
+                             oceanHeatTransferVelocity,&
+                             oceanSaltTransferVelocity,&
+                             interfacePressure,&
+                             iceTemperature,&
+                             iceTemperatureDistance, &
+                             iceFloatingMask, &
+                             outInterfaceSalinity,&
+                             outInterfaceTemperature,&
+                             outFreshwaterFlux,&
+                             outOceanHeatFlux,&
+                             outIceHeatFlux,&
+                             gsize)
+
+    !do n=1,gsize
```

Comment:
More commented-out code to remove.

Reply:
Fixed in 733a46925b5580e5888bdfa417add9916e7cf2df